### PR TITLE
Lasagna concept exercise

### LINKED ISSUE #565: lasagna: remove TODO from `introduction.md`
The `docs/introduction.md` document contains the following Markdown:

```
## Annotations

<!-- TODO: These are not concepts/concept headings, but what are they? -->

### Types

...

### Comments
```

I think want we want here is to have:

```
## Types

...

## Comments
```

### PR DESCRIPTION
**Pull request by glennj**
Reworked the `functions` concept a bit.
Added `integer-introduction` to fulfill the lasagna design doc. 
Moved comments from functions to its own concept.

I added a `config.json.wip` file to track the syllabus.
Closes #565

### CODE REVIEW

**Comment by glennj** [comment]
`configlet generate` doesn't work due to the "wip" dirnames, so the exercise's introduction.md might be incorrect.

**Comment by glennj** [comment]
@ErikSchierboom FYI, I don't have privs to add reviewers, and you're not in the julia team.

**Review by ErikSchierboom — APPROVED** [review]
LGTM! I'll let a Julia maintainer look at the julia bits

File: config.json.wip
Comment (SaschaMann) [inline config.json.wip:1]: There's already a wip `config.json` in `concepts.wip`, how does this one differ? https://github.com/exercism/julia/blob/main/concepts.wip/config.json

File: concepts.wip/integer-introduction/about.md
Comment (SaschaMann) [inline concepts.wip/integer-introduction/about.md:3]: What's the unusual way?

File: concepts.wip/integer-introduction/about.md
Comment (SaschaMann) [inline concepts.wip/integer-introduction/about.md:66]: Perhaps we can add an admonition here that in the REPL and most Julia editing environments, this can be typed as `\div<tab>`. There are similar hints in other files if you want an example.

File: concepts.wip/integer-introduction/about.md
Comment (SaschaMann) [inline concepts.wip/integer-introduction/about.md:38]:
```suggestion
The standard prefix and infix operations are available: `+`, `-`, `*`, `%`
```

since it's listed below

File: concepts.wip/integer-introduction/introduction.md
Comment (SaschaMann) [inline concepts.wip/integer-introduction/introduction.md:1]: Same as above

File: exercises/concept.wip/lasagna/lasagna.jl
Comment (SaschaMann) [inline exercises/concept.wip/lasagna/lasagna.jl:1]: I'm not a fan of these kind of not implemented errors as [they are an anti-pattern in Julia](https://www.oxinabox.net/2020/04/19/Julia-Antipatterns.html) and I'm generally against introducing anti-pattern as part of teaching even if they're not critical in this particular situation. So I'd prefer removing the errors and leaving them as empty functions.

There's previous discussion to be found somewhere that ended up in removal of almost all stubs.

Personally I think the comments are redundant and might lead to divergence between instructions and stub in the long run but I don't mind too much.

**Review by SaschaMann — COMMENTED** [review]
Thank you!

File: exercises/concept.wip/lasagna/lasagna.jl
Comment (SaschaMann) [reply exercises/concept.wip/lasagna/lasagna.jl:1]: 198ad16f39a56054962112497ea95646bf14a4ce solved these while I was typing :D

File: config.json.wip
Comment (glennj) [reply config.json.wip:1]: Oops missed that. I'll merge mine into the existing one.

File: concepts.wip/integer-introduction/about.md
Comment (glennj) [reply concepts.wip/integer-introduction/about.md:3]: Well, the target audience is programmers, so I'm assuming they've done it before in some language. Is there a more descriptive way to describe how to write an integer?

File: exercises/concept.wip/lasagna/lasagna.jl
Comment (glennj) [reply exercises/concept.wip/lasagna/lasagna.jl:1]: Erik pointed me to the elixir stub which is much more minimal but still lays out the tasks.

Comment (SaschaMann) [reply exercises/concept.wip/lasagna/lasagna.jl:1]: Yep, that's fine

**Comment by ErikSchierboom** [comment]
@glennj Did you intend for the changes to remain in the `*.wip` directory? We could have this exercise be listed as a regular concept exercises. As `status.concept_exercises` is `false`, it won't show up for anyone except for maintainer IIRC.

**Comment by ErikSchierboom** [comment]
Oh, and thanks a ton for working on this! 🥳

**Pull request merged**
